ideascube/serveradmin/catalog.py

```diff
@@ -604,9 +606,19 @@ def list_nothandled(self, ids):
 
         return sorted(pkgs, key=attrgetter('id'))
 
+    def _update_displayed_packages_on_home(self, *, to_remove_ids=None, to_add_ids=None):
```

Comment:
I know what using `*` in a function signature does. My question was **why** you would use it here.

Only local stuff should use this method, so when calling it we'd have the source to see what the parameters should be. (and it is private, so it's not API, so it's fine to have to read the code to understand it IMHO)

Plus, I know it's possible to call a function specifying the kwargs as positional arguments, but that's just terribly bad taste and should always be avoided. I know I wouldn't accept merging something like this in a review:

```
def foo(bar=None):
    ...

foo('value')
```

(aside: I've always thought that Python itself should enforce passing kwargs as kwargs, that the snippet I wrote above should raise an error)

So all in all, I don't really see the point of the `*`, and I think it has too much potential for confusion. It's just too easy to mistake it for a `_` and think the function takes a positional argument that it doesn't use. It is also so rarely used, and so hard to search for (try searching for "python *" on a search engine :stuck_out_tongue_winking_eye:) that it certainly would cause trouble to a new developer.

Basically what I'm saying is that I dislike this `*` construction and I think it shouldn't be used because the benefits are negligible compared to the benefits, as long as we are careful to always pass kwargs as kwargs and never as positional args.

But that's just my opinion, which is why I asked you **why** you used it, because it is very possible that I'm wrong here and I was interested in your opinion, in case you had arguments that could make me change my mind. :smiley: